Develop

QUDUSKUNLE opened the pull request.

QUDUSKUNLE committed:
Merge pull request #8 from QUDUSKUNLE/master

master into develop

Commit:
[Ch-#153279622] Write UserController test

- Move UserController mockData to userData file

Commit:
[Ch-#153279622] Fix database configuration error

Commit:
[Ch-#153279622] Rename verifyToken to adhere to Airbnb

QUDUSKUNLE committed:
Merge pull request #9 from QUDUSKUNLE/ch-server-side-test-153279622

Ch server side test 153279622

Commit:
[Feature #153245387]
Create reset password route
- Move dbConfig to .env
- Create update password route
- Create Update reset password email template
- Write test for reset password route
- Write test for update password route

Commit:
[Feature #153245387] Fix mongodb connection error

Commit:
[Feature #153245387] create .coveralls.yml
- update .travis.yml with Code Climate token

Commit:
[Feature #153245387] Install cat dependency for coverage

Commit:
[Feature #153245387] Obtain coverage

QUDUSKUNLE committed:
Merge pull request #11 from QUDUSKUNLE/ft-reset-password-153245387

Ft reset password 153245387

Commit:
[Feature #153245419] Craft createIdea endpoint
- Craft edit idea endpoint
- Crafet delete idea endpoint

Commit:
[Feature #153245419] Write ideaController.spec.js test
- Update mockData
- Write userController.spec.js test

Commit:
[Feature #153245419] Update CodeClimate.yml
- Update .eslintrc.json
- Update .travis.yml with codeclimate repo token
-

Commit:
[Feature #153245419] Write functions to validate requests
- Update UserController.js
- Rename email environment variables
- Remove redundant code from the server.js file

QUDUSKUNLE committed:
Merge pull request #13 from QUDUSKUNLE/ft-create-ideas-153245419

Ft create ideas 153245419

Commit:
[Feature #engines:]

Commit:
[Feature #153245722] Correct codeclimate.yml configuration error

Commit:
[Chore #153245722] Update .eslintrc.json
- Install some dependencies
- Update README

Commit:
[Feature #153245722] Craft filter idea endpoint
- Create searchIdea function
- Create Procfile file

Commit:
[Feature #153245722] Write test models test

QUDUSKUNLE committed:
Merge pull request #14 from QUDUSKUNLE/ft-filter-ideas-153245722

Ft filter ideas 153245722

Commit:
[Chore #153246043] Obtain badges

johadi commented on server/routes/index.js:
```diff
@@ -1,7 +1,7 @@
-import Validator from '../middlewares/Validator';
-import ValidateToken from '../middlewares/ValidateToken';
+import validateRequest from '../middlewares/validateRequest';
+import verifyToken from '../middlewares/verifyToken';
```
Nice. Though this is good .I just want let you know there is one cool way of routing too that can save your time from duplication of writing codes. 
```
Router.route('/api/signup')
      .get(getSignupController)
      .post(postSignupController)
// You can see how clean that is instead of doing the following
Router.get('/api/signup', postSignupController)
Router.post('/api/signup', getPostController);
````
I just to bring it to your notice, you didn't really have such use case here

Commit:
[Chore #153246107] Configure webpack for both development and production
- install necessary dependencies

Commit:
[Chore #153246107] Set up webpack for development

QUDUSKUNLE committed:
Merge pull request #15 from QUDUSKUNLE/ch-link-webpack-to-server-153246107

Ch link webpack to server 153246107

Commit:
[Chore #153279622] Write more controllers test

Commit:
[Chore #153279622] Add production configuration

Commit:
[Chore #153279622] Write pagination test

Commit:
[Chore #153279622] Write test for Filter ideas by category

QUDUSKUNLE committed:
Merge pull request #16 from QUDUSKUNLE/ch-server-side-test-153279622

Ch server side test 153279622

Commit:
[Chore 153245231] Implement landing page

Commit:
[Feature #153245231]
- Create AppActions file
- Craft signUpUser action
- Define Application Store
- Create AppConstants file

Commit:
[Feature #153245231]
- Design a template to display error message
- implement signup feature
- setAuthToken to header for handle request

Commit:
[Feature #153245231] Remove all console logs

QUDUSKUNLE committed:
Merge pull request #17 from QUDUSKUNLE/ft-client-register-153245231

Ft client register 153245231

Commit:
[Heroku Build error] Fix heroku build error

QUDUSKUNLE committed:
Rename Appdispatcher.js to AppDispatcher.js

Commit:
[Feature #153245311] Implement log in functionality
- Design custom alert to catch error messages
- Craft log in user method in AppActions

Commit:
[Feature #153245311)] Implement reset password functionality
- Customiz alert message when reset email is sent
- craft resetPassword method in the AppActions

QUDUSKUNLE committed:
Merge pull request #18 from QUDUSKUNLE/ft-client-login-153245311

Ft client login 153245311

Commit:
[Feature #153245387] Add file loader to webpack

Commit:
[Feature #153245387] Redirect to dashboard when users signed in

Commit:
[Feature #153245387] Add Routes component
- Abstract routes into a signle function Routes
- Design DashBoard Page

QUDUSKUNLE committed:
Merge pull request #19 from QUDUSKUNLE/ft-client-reset-password-153245387

Ft client reset password 153245387

Commit:
[Feature #153245419] Add create idea feature
- implement routing to dashboard when user sign up or sign in successfully

Commit:
[Feature #153245419] Add fetchPublic Ideas endpoint
- Add endpoint to fetch all ideas created by a user
- Add endpoint to fetch a single idea

- Write test for the endpoints

Commit:
[Feature #153245419-client] Implement create Idea feature

QUDUSKUNLE committed:
Merge pull request #20 from QUDUSKUNLE/ft-client-create-idea-153245419

Ft client create idea 153245419

Commit:
[Feature #153245722] Add react-paginate to handle pagination

Commit:
[Feature #153245722] Fix Jquery conflict

Commit:
[Feature #153245722] Craft fetch Idea by Category Action
- Create a method in the Store that handles fetchByCategory action

Commit:
[Feature #153245722] Re-implement registration and log in

Commit:
[Feature #153245722] Re craft filterByCategory API call
- Add pagination method

Commit:
[Feature #153245722] implement filter idea by category
- [#finish]

Commit:
[Feature #153245722] Write test for filterCategory in the ideaController

Commit:
[Feature #153245722] Remove JQuery from webpack

QUDUSKUNLE committed:
Merge pull request #21 from QUDUSKUNLE/ft-client-filter-idea-153245722

Ft client filter idea 153245722

Commit:
[Feature #153245705] Re craft search Ideas endpoint
- handle pagination for search ideas endpoint

Commit:
[Feature #153549714] Add MyIdeas Componenet
- Declare MyIdeas Component in Routes
- Craft myIdeas action
- Remove dead codes from DashBoard.jsx
- Craft a method in the store to handle MyIdeas page events

Commit:
[Feature #153549714] Add view more to ideas
- Re implement userIdeas API
- style view_more icons

Commit:
[Feature #153549714] Ignore EditIdea Component

Commit:
[Feature #153549714] Fix Broken test

QUDUSKUNLE committed:
Merge pull request #22 from QUDUSKUNLE/ft-client-myIdeas-153549714

Ft client my ideas 153549714

Commit:
merge branch 'develop' of https://github.com/QUDUSKUNLE/Ideabox into ft-client-search-ideas-153245705

Commit:
[Feature #153245705] Fix searchIdea endpoint

Commit:
[Feature #153245705] Add search-ideas functionality
- Write test for search idea endpoint

QUDUSKUNLE committed:
Merge pull request #23 from QUDUSKUNLE/ft-client-search-ideas-153245705

Ft client search ideas 153245705

Kenec commented on client/src/actions/AppActions.js:
```diff
@@ -150,5 +150,22 @@ export default {
+        })
+    );
   }
 };
```
Good job!
In your signUpUser  and  LoginUser functions, I do not see the need of getting the token from the localStorage since it is available in the response. I think the code will execute faster when we don't call processes that we can avoid.

In LoginUser function, I don't think storing the username in the localStorage is ideal. Since you already have a token saved in the localStorage, just encode the username as part of the token and decode it when you need to make use of it. Same applies to isAuthenticated. We don't want to make it easy for a malicious user to have unauthorized access to our app.

Commit:
[Feature #153245691] - Add Update Profile functionality
- Craft update profile Action
- Abstract localStorage set into a function
- Dispatch UpdateProfile Action to Store
[#Finish]

Commit:
[Feature #153245691] - Fix failed test

QUDUSKUNLE committed:
Merge pull request #24 from QUDUSKUNLE/ft-client-edit-profile-153245691

Ft client edit profile 153245691
